src/verbs/queue_pair.rs

```diff
@@ -176,13 +177,22 @@ pub trait QueuePair {
     type Guard<'g>: PostSendGuard
     where
         Self: 'g;
-    fn start_post_send<'qp, 'g>(&'qp mut self) -> Self::Guard<'g>
-    where
-        'qp: 'g;
+    fn start_post_send(&mut self) -> Self::Guard<'_>;
+
+    fn start_post_recv(&mut self) -> PostRecvGuard<'_> {
+        PostRecvGuard {
+            qp: unsafe { self.qp() },
+            wrs: Vec::new(),
```

Comment:
We should align the logic in `BasicPostSendGuard` using `Vec::new()` as well? Maybe in a separate commit.

Reply:
Yeah, I think `Vec::new()` is better when we want to create an empty vector.